ctf_stats: Add map name to match summary

ClobberXD opened the pull request:
![screenshot_20191109_173245](https://user-images.githubusercontent.com/36130650/68528333-4a114b80-0317-11ea-8d93-a73f5f2c482c.png)

Opinions and thoughts regarding the placement of the map name is appreciated. Obviously, what looks good on my screen could look completely borked on another screen. That's the inherent limitation of formspecs; hopefully `real_coordinates` would help improve consistency of formspecs across different screens.

Tested; works. Pretty trivial to review and test.

ClobberXD commented:
Added `Map: ` before the map name

Thomas--S commented on mods/ctf/ctf_stats/init.lua:
```diff
@@ -215,6 +215,12 @@ ctf_match.register_on_new_match(function()
+
+	-- The delay is required to prevent a race condition, because ctf_map
+	-- uses the same on_new_match callback for map initialization
+	minetest.after(1, function()
```
To be honest I strongly oppose such a time-based approach.

Strictly speaking, this still adds kind of a race condition because it cannot be guaranteed that the other function takes less than a second.

The clean way to to this is to allow registering callbacks in the ctf_map mod that are run when the new map name is set there.

ClobberXD replied:
1s is far too much to create a race condition, but I can see where you're going with this.

> The clean way to to this is to allow registering callbacks in the ctf_map mod that are run when the new map name is set there.

Fair point. I'll do just that.

ClobberXD replied:
Done

ClobberXD commented:
I've added a new callback `ctf_map.register_on_map_loaded`, that'll be executed when a map has fully finished loading. I've removed the `minetest.after(1, ...)`, and replaced it with the callback. Note that directly adding `ctf_map` as a dependency of `ctf_stats` will create a cyclic dependency, so I resorted to using `minetest.after(0, ctf_map.register_on_map_loaded)` instead. PR tested, works perfectly.

Note that there's a debug message for easier testing, and should be removed before merge.

ClobberXD commented:
Debug message removed. Feel free to merge this if you like.
